### ISSUE TITLE
Breaks shell redirection and pipes

### ISSUE THREAD

**Issue by ben-cohen-xs** [bogado/file-line#82]
The `file-line` plugin breaks shell redirection: the command below should start `vim` with a buffer containing its manpage, but instead `vim` shows an empty buffer with a name like "/proc/65432/fd/63".
```
vim <(man vim)
```

The plugin breaks it because it tries to re-read the buffer, which is not possible from a pipe.  You can fix the example above like this, though it would be better to do a more general check for pipes (and other special files) because they can also only be read once.
```
90c90
< if match(expand("%:p"), "^/proc/.*/fd/.*$") < 0 && match(expand("%:p"), "^/dev/fd/.*$") < 0 && !isdirectory(expand("%:p"))
---
> if !isdirectory(expand("%:p"))
```

**Comment by xim** [comment]
We could just add a test for `!filereadable(expand("%"))` so startup isn't called if there's already something readable there.

**Comment by ben-cohen-xs** [comment]
Yes, this works for me and is neater than the change in my previous comment:
```
if !filereadable(expand("%:p")) && !isdirectory(expand("%:p"))
```

**Comment by kaddkaka** [comment]
Any progress on this issue?

**Comment by chrysn** [comment]
Adding the filereadable test also fixed my issue where editing `.gpg` files (with transparent de- and encryption) broke after adding this plugin. (It comes at the cost of not jumping to the right line in `.gpg` files, but that's rarely where my compiler is reporting line numbers of errors for :-) ).
